hypervisor/include/arch/x86/apicreg.h

```diff
@@ -393,42 +405,29 @@ union ioapic_rte {
 /*
  * fields in the IO APIC's redirection table entries
  */
-#define IOAPIC_RTE_DEST_SHIFT  	56U
-/* broadcast addr: all APICs */
-#define IOAPIC_RTE_DEST_MASK	0xff00000000000000UL
-
-#define IOAPIC_RTE_RESV	0x00fe0000UL	/* reserved */
-
-#define IOAPIC_RTE_INTMASK	0x00010000UL	/* R/W: INTerrupt mask */
-#define IOAPIC_RTE_INTMCLR	0x00000000UL	/*       clear, allow INTs */
-#define IOAPIC_RTE_INTMSET	0x00010000UL	/*       set, inhibit INTs */
-
-#define IOAPIC_RTE_TRGRMOD	0x00008000UL	/* R/W: trigger mode */
-#define IOAPIC_RTE_TRGREDG	0x00000000UL	/*       edge */
-#define IOAPIC_RTE_TRGRLVL	0x00008000UL	/*       level */
+#define IOAPIC_RTE_INTMCLR	0x0U	/*       clear, allow INTs */
+#define IOAPIC_RTE_INTMSET	0x1U	/*       set, inhibit INTs */
 
-#define IOAPIC_RTE_REM_IRR	0x00004000UL	/* RO: remote IRR */
+#define IOAPIC_RTE_TRGREDG	0x0U	/*       edge */
+#define IOAPIC_RTE_TRGRLVL	0x1U	/*       level */
 
-#define IOAPIC_RTE_INTPOL	0x00002000UL /*R/W:INT input pin polarity*/
-#define IOAPIC_RTE_INTAHI	0x00000000UL	/*      active high */
-#define IOAPIC_RTE_INTALO	0x00002000UL	/*      active low */
+#define IOAPIC_RTE_REM_IRR	0x1U	/* RO: remote IRR */
 
-#define IOAPIC_RTE_DELIVS	0x00001000UL	/* RO: delivery status */
+#define IOAPIC_RTE_INTAHI	0x0U	/*      active high */
+#define IOAPIC_RTE_INTALO	0x1U	/*      active low */
```

Comment:
I need you to add additional comments for the MACRO of a field in a union. Please explicitly say it is for which field...

BTW, can we define the MACRO inside the union?  Just for brainstorming...

Reply:
and close to the field definition.

Reply:
I was wondering if mixing struct/union declarations with macro definitions will eventually confuse doxygen when we add documents to these fields. @dbkinder would you please help check this? Thanks!

Reply:
In this case, how about this way:
1: Those MACROs are defined right after the union  is defined.
2: For IOAPIC_RTE_INTAHI, let us do like:
IOAPIC_RTE_INTAHI           /* intr_polarity: active high */